src/plugins/intel_gpu/tests/unit/test_cases/slice_gpu_test.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "test_utils.h"
+
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <intel_gpu/primitives/slice.hpp>
+#include <intel_gpu/primitives/data.hpp>
+
+#include <random>
+#include <algorithm>
+#include <vector>
+
+using namespace cldnn;
+using namespace ::tests;
+
+namespace {
+
+namespace helpers {
+
+// data_types type traits:
+template<typename T>
+data_types ToDataType();
+
+template<>
+data_types ToDataType<float>() {return data_types::f32;}
+
+template<>
+data_types ToDataType<int32_t>() { return data_types::i32; }
+
+template<>
+data_types ToDataType<int64_t>()  { return data_types::i64; }
+
+// Generates buffer of the length of shape_size(shape).
+template<typename T>
+std::vector<T> GenInput(const ov::PartialShape& shape) {
+    const size_t size = ov::shape_size<ov::Shape>(shape.get_shape());
+    std::vector<T> result;
+    for (size_t i = 0; i < size; i++)
+        result.push_back(static_cast<T>(i));
+    return result;
```

Comment:
```suggestion
    const auto size = ov::shape_size(shape.get_shape());
    auto result = std::vector<T>(size);
    std::iota(result.begin(), result.end(), T{0});
    return result;
```
If T is fundamental type then should be simpler and result vector would allocated once.